contributing/development/cpp_usage_guidelines.rst

```diff
@@ -53,18 +53,69 @@ as Godot provides its own data types (among other things).
+| ``RBSet``              | ``std::set``             | Uses a `red-black tree <https://en.wikipedia.org/wiki/Red-black_tree>`__            |
+|                        |                          | for faster access.                                                                  |
++------------------------+--------------------------+-------------------------------------------------------------------------------------+
+| ``HashMap``            | ``std::map``             | **Use this as the "default" map type.** Iteration is different from ``std::map``    |
+|                        |                          | (``HashMap`` is insertion-ordered, while ``std::map`` is key-ordered).              |
++------------------------+--------------------------+-------------------------------------------------------------------------------------+
+| ``AHashMap``           | ``std::map``             | Array-based implementation of a hash map.                                           |
++------------------------+--------------------------+-------------------------------------------------------------------------------------+
+| ``OAHashMap``          | ``std::unordered_map``   | Does not preserve insertion order. Generally faster than ``HashMap`` and            |
+|                        |                          | ``AHashMap``.                                                                       |
++------------------------+--------------------------+-------------------------------------------------------------------------------------+
+| ``Dictionary`` 📜      | ``std::map``             | Keys and values can be of any Variant type. No static typing is imposed.            |
++------------------------+--------------------------+-------------------------------------------------------------------------------------+
+| ``TypedDictionary`` 📜 | ``std::map``             | Equivalent to ``Dictionary`` but with static typing for its keys and values.        |
++------------------------+--------------------------+-------------------------------------------------------------------------------------+
+| ``Pair``               | ``std::pair``            | Stores a single key-value pair.                                                     |
++------------------------+--------------------------+-------------------------------------------------------------------------------------+
+
+\*: For certain use cases, this alternative datatype is more performant or has
+characteristics that are more suitable for the task at hand. However, it is not
+always a direct replacement for the recommended datatype. Ask for advice if you
+are unsure.
 
 .. note::
 
-    Godot also has a List datatype (which is a linked list). While List is already used
-    in the codebase, it typically performs worse than other datatypes like Vector
-    and Array. Therefore, List should be avoided in new code unless necessary.
+    Godot also has a ``List`` datatype (which is a linked list). While ``List``
```

Comment:
It's probably fine to move `List` into the table, and just put the usage disclaimer in the comment box. Slow or not, it _is_ a regular collection type.